lua/org-roam/database/loader.lua

```diff
@@ -348,6 +348,8 @@ function M:load_file(opts)
                         insert_new_file_into_database(db, file, {
                             force = opts.force or file.metadata.changedtick ~= changedtick,
                         })
+
+                        require("orgmode").files:load(true)
```

Comment:
@seflue line 329 should already add this file. 
This might be an issue in orgmode, because we do early return here https://github.com/nvim-orgmode/orgmode/blob/master/lua/orgmode/files/init.lua#L62-L64.

Can you try comment out that early return, remove this line, and see if it works then?

Reply:
@kristijanhusak Thanks for considering a more optimal solution. I also have the feeling, that something can be improved in loading org-files. 

While you are already reflecting on this topic I want to bring in another thought: For Telescope-orgmode I always wished for incremental loading capabilities of orgfiles. Like a stream, so user get's immediate feedback as soon as the first org file has been parsed and the results can be updated with each file parsed. Telescope itself is built to handle such cases, but orgmode currently does not support this.

Is this something you would need to factor into your considerations of optimizing file loading or should I create another issue for this?

Reply:
> While you are already reflecting on this topic I want to bring in another thought: For Telescope-orgmode I always wished for incremental loading capabilities of orgfiles. Like a stream, so user get's immediate feedback as soon as the first org file has been parsed and the results can be updated with each file parsed. Telescope itself is built to handle such cases, but orgmode currently does not support this.

I don't plan to add a stream like loading, because then I would have to worry about the pending files when showing the agenda and similar. There is caching in place so loading is slow only once.
Even this `load(true)` function should be fairly fast in this scenario because only files that are not already loaded will be loaded, but I'll still see to add a method that will load the file and add it to path, but only if it belongs to path.

Reply:
If you add this method to load a single file, I think, I could add this stream-based approach myself in Telescope-orgmode. For me it's just a matter of a flexible API to provide plugins the possibility to do, what they need. That you don't want to implement a stream-based loading directly in orgmode makes sense.

Reply:
@chipsenkbeil do you recall if this `add_to_paths` is needed to just load the file to `files` table, or we also need it to append it to path?

Currently, `add_to_paths` will load the file, put it in `OrgFiles.files` table, and also append the filename to `OrgFiles.paths` if the file does not belong to the defined path wildcard.

So for example, if my org-roam config is `~/roam/**/*`, orgmode org_agenda_files is `~/org`, and I do `:add_to_paths('~/some/other/path/file.org')` in any of those file loaders, this will add the file to the `paths` for the session, even though that might not be desired.
If a file that is already part of the path wildcard, for example `:add_to_paths('~/roam/myfile.org')` with a roam file loader, we are not adding it to `paths`, since it will be loaded anyway.

My question is: Is it ok if we do not append the filename to `OrgFiles.paths` in `add_to_paths`? Does org-roam always creates files that are defined in the path wildcard (in my example, `~/roam/**/*`)?

Reply:
Org roam takes a directory path, not a glob, and always creates node files within that directory. Nowhere else.

To be compatible with your plugin, I convert the directory to a glob doing a simple path join: `join_path("path/to/roam", "**", "*")`.

I'd have to go back to read why the function was added.

Reply:
@kristijanhusak I tried to skim through the code to refresh myself.

With `OrgFiles`, I directly access the `all_files` field, and make use of these functions:

1. filenames
2. load_file
3. add_to_paths
4. load

I pass `OrgFiles` to things like `OrgCapture`. I don't think I access `paths` directly.

Reply:
>  and always creates node files within that directory

This answers my question, thanks!

@seflue let me do some updates on the orgmode side and I'll let you know what to change here.

@chipsenkbeil If you need some helper function to avoid accessing properties directly let me know, we can add something.